### PR TITLE
2nd enemy sprite

### PR DESCRIPTION
**Pull request by HifenX10**

### CODE REVIEW

File: Assets/Scripts/Mechanics/EnemyController.cs [hunk 1]
```diff
@@ -14,76 +16,141 @@ namespace Platformer.Mechanics
     [RequireComponent(typeof(AnimationController), typeof(Collider2D))]
     public class EnemyController : MonoBehaviour, IEnemy
     {
+        protected bool facingRight = false;
+
+        public Component ComponentFacing;
+        public float ContactDammageCoolDownSeconds => 2;
+        public virtual float BounceAmount => 7;
+        public virtual bool HurtByJump => true;
+        public virtual int ContactDammage => 2;
```
Comment (marcsoiferman) [inline Assets/Scripts/Mechanics/EnemyController.cs:25]: Spelling mistake

File: Assets/Scripts/Mechanics/EnemyController.cs [hunk 2]
```diff
@@ -14,76 +16,141 @@ namespace Platformer.Mechanics
     [RequireComponent(typeof(AnimationController), typeof(Collider2D))]
     public class EnemyController : MonoBehaviour, IEnemy
     {
+        protected bool facingRight = false;
+
+        public Component ComponentFacing;
+        public float ContactDammageCoolDownSeconds => 2;
```
Comment (marcsoiferman) [inline Assets/Scripts/Mechanics/EnemyController.cs:22]: Spelling mistake

File: Assets/Scripts/Mechanics/KinematicObject.cs [hunk 3]
```diff
@@ -87,6 +87,9 @@ public void Bounce(Vector2 dir)
         /// <param name="position"></param>
         public void Teleport(Vector3 position)
         {
+            //if (body is null)// I dont' know why body is sometimes null, but its an issue.
+            //    return;
+
```
Comment (marcsoiferman) [inline Assets/Scripts/Mechanics/KinematicObject.cs:92]: Can we delete this?

File: Assets/Scripts/Mechanics/SniperJoeController.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    public override float BounceAmount => 3;
+    public override bool HurtByJump => false;
+    public override int ContactDammage => 3;
+    public SniperJoeAnimationController animationController;
+    public Rigidbody2D body;
+
+    public EnemyStance mode { get; set; }
+    private float modeDeltaTime; 
+
+    SniperJoeWeapon weapon;
```
Comment (marcsoiferman) [inline Assets/Scripts/Mechanics/SniperJoeController.cs:29]: Access modifier!

File: Assets/Weapons/EnemyBlaster.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Assets.Weapons
+{
+    class EnemyBlaster
+    {
+    }
+}
```
Comment (marcsoiferman) [inline Assets/Weapons/EnemyBlaster.cs:12]: Why is this empty?

File: Assets/Scripts/Mechanics/KinematicObject.cs (hunk 3 above)
Comment (HifenX10) [reply Assets/Scripts/Mechanics/KinematicObject.cs:92]: removed

File: Assets/Scripts/Mechanics/SniperJoeController.cs (hunk 4 above)
Comment (HifenX10) [reply Assets/Scripts/Mechanics/SniperJoeController.cs:29]: added

File: Assets/Weapons/EnemyBlaster.cs (hunk 5 above)
Comment (HifenX10) [reply Assets/Weapons/EnemyBlaster.cs:12]: removed.

File: Assets/Scripts/Mechanics/EnemyController.cs (hunk 2 above)
Comment (HifenX10) [reply Assets/Scripts/Mechanics/EnemyController.cs:22]: corrected

File: Assets/Scripts/Mechanics/EnemyController.cs (hunk 1 above)
Comment (HifenX10) [reply Assets/Scripts/Mechanics/EnemyController.cs:25]: corrected.

**Pull request merged**